templates/cicd/pipeline_cfn.yml

```diff
@@ -179,6 +179,22 @@ Resources:
               - arn:aws:iam::{{$stage.AccountID}}:role/{{$.AppName}}-{{$stage.Name}}-EnvManagerRole{{end}}
       Roles:
         - !Ref PipelineRole
+  {{$length := len .Stages}}{{if gt $length 0}}{{range $index, $stage := .Stages}}
+  BuildTestCommands{{$index}}:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Name: BuildTestCommands{{$index}}
+      EncryptionKey: !ImportValue {{$.AppName}}-ArtifactKey
+      ServiceRole: !GetAtt BuildProjectRole.Arn
+      Artifacts:
+        Type: NO_ARTIFACTS
+      Environment:
+        Type: LINUX_CONTAINER
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
+        ComputeType: BUILD_GENERAL1_SMALL
+      Source:
+        Type: NO_SOURCE
+        BuildSpec: "version: 0.2\nphases:\n  build:\n    commands: [{{range $index, $command := $stage.TestCommands}}{{if $index}},{{end}}\"{{$command}}\"{{end}}]"{{end}}{{end}}
```

Comment:
Is the trailing comma on the last command okay to have?

Reply:
i think there's no trailing comma on the last command🙂

Reply:
yea, you're right :P

Reply:
There shouldn't be a trailing comma, because the comma precedes non-zero-indexed commands. Example output from current code:
```
Source:
        Type: NO_SOURCE
        BuildSpec: "version: 0.2\nphases:\n  build:\n    commands: [\"echo 'running tests'\",\"cat README.md\"]"
```

Reply:
> i think there's no trailing comma on the last command🙂

Oops missed this comment earlier! :)